providers/realtimeregister/realtimeregisterProvider.go

```diff
@@ -0,0 +1,328 @@
+	case "NAPTR":
+		_ = recordConfig.SetTargetNAPTRString(record.Content)
+	case "SRV":
+		parts := strings.Fields(record.Content)
+		weight, _ := strconv.ParseUint(parts[0], 10, 16)
+		port, _ := strconv.ParseUint(parts[1], 10, 16)
+		_ = recordConfig.SetTargetSRV(uint16(record.Priority), uint16(weight), uint16(port), addTrailingDot(parts[2]))
+	case "CAA":
+		_ = recordConfig.SetTargetCAAString(record.Content)
+	case "SSHFP":
+		_ = recordConfig.SetTargetSSHFPString(record.Content)
+	case "TLSA":
+		_ = recordConfig.SetTargetTLSAString(record.Content)
+	case "DS":
+		_ = recordConfig.SetTargetDSString(record.Content)
+	case "LOC":
+		_ = recordConfig.SetTargetLOCString(domain, record.Content)
+	default:
+		_ = recordConfig.SetTarget(record.Content)
+	}
+	return recordConfig
+}
+
+func toRecord(recordConfig *models.RecordConfig) Record {
+	record := &Record{
+		Type:     recordConfig.Type,
+		Name:     recordConfig.NameFQDN,
+		Content:  removeTrailingDot(recordConfig.GetTargetField()),
+		TTL:      int(recordConfig.TTL),
+		Priority: int(recordConfig.MxPreference),
+	}
+
+	switch rtype := recordConfig.Type; rtype {
+	case "SRV":
+		record.Priority = int(recordConfig.SrvPriority)
+		record.Content = fmt.Sprintf("%d %d %s", recordConfig.SrvWeight, recordConfig.SrvPort, record.Content)
+	case "NAPTR", "SSHFP", "TLSA", "CAA":
+		record.Content = recordConfig.GetTargetCombined()
+	case "TXT":
+		record.Content = fixBackslashesAndDoubleQuotes(record.Content, false)
+	case "DS":
+		record.Content = fmt.Sprintf("%d %d %d %s", recordConfig.DsKeyTag, recordConfig.DsAlgorithm,
+			recordConfig.DsDigestType, strings.ToUpper(recordConfig.DsDigest))
+	case "MX":
+		if record.Content == "." {
+			record.Priority = -1
+		}
+	case "LOC":
+		parts := strings.Fields(recordConfig.GetTargetCombined())
+		degrees1, _ := strconv.ParseUint(parts[0], 10, 32)
+		minutes1, _ := strconv.ParseUint(parts[1], 10, 32)
+		degrees2, _ := strconv.ParseUint(parts[4], 10, 32)
+		minutes2, _ := strconv.ParseUint(parts[5], 10, 32)
+		altitude, _ := strconv.ParseFloat(strings.Split(parts[8], "m")[0], 64)
+		size, _ := strconv.ParseFloat(strings.Split(parts[9], "m")[0], 64)
+		hp, _ := strconv.ParseFloat(strings.Split(parts[10], "m")[0], 64)
+		vp, _ := strconv.ParseFloat(strings.Split(parts[11], "m")[0], 64)
+		record.Content = fmt.Sprintf("%d %d %s %s %d %d %s %s %.2fm %.2fm %.2fm %.2fm",
+			degrees1, minutes1, parts[2], parts[3], degrees2, minutes2,
+			parts[6], parts[7], altitude, size, hp, vp,
+		)
+	}
+
+	return *record
+}
+
+func (api *realtimeregisterApi) EnsureZoneExists(domain string) error {
+	exists, err := api.zoneExists(domain)
+	if err != nil {
+		return err
+	}
+	if exists {
+		return nil
+	}
+
+	return api.createZone(domain)
+}
+
+func removeTrailingDot(record string) string {
+	if record == "." {
+		return record
+	}
+	return strings.TrimSuffix(record, ".")
+}
+
+func fixBackslashesAndDoubleQuotes(name string, inverse bool) string {
```

Comment:
Rather than a flag `inverse`, create 2 functions, one that encoded and one that decodes.

It's considered bad style to have a parameter that makes a function do 2 relatively unrelated functions.  All the callers of this function set inverse to a constant (either `true` or `false`), which indicates this is two different functions in one.